src/wp-admin/includes/image.php

```diff
@@ -445,16 +557,30 @@ function _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id ) {
 	}
 
 	if ( method_exists( $editor, 'make_subsize' ) ) {
+		// Make the primarey mime type sub sized images
 		foreach ( $new_sizes as $new_size_name => $new_size_data ) {
-			$new_size_meta = $editor->make_subsize( $new_size_data );
-
+			$new_size_meta = $editor->make_subsize( $new_size_data, $primary_mime_type );
 			if ( is_wp_error( $new_size_meta ) ) {
 				// TODO: Log errors.
 			} else {
 				// Save the size meta value.
 				$image_meta['sizes'][ $new_size_name ] = $new_size_meta;
+				if ( ! isset( $image_meta['sizes'][ $new_size_name ]['sources'] ) ) {
+					$image_meta['sizes'][ $new_size_name ]['sources'] = array();
+				}
+				$image_meta['sizes'][ $new_size_name ]['sources'][ $primary_mime_type ] = _wp_get_sources_from_meta( $new_size_meta );
 				wp_update_attachment_metadata( $attachment_id, $image_meta );
 			}
+
+			// Save additional mime types for this size.
+			foreach ( $additional_mime_types as $output_mime_type ) {
+				$new_size_meta = $editor->make_subsize( $new_size_data, $output_mime_type );
+
+				if ( ! is_wp_error( $new_size_meta ) ) {
+					$image_meta['sizes'][ $new_size_name ]['sources'][ $output_mime_type ] = _wp_get_sources_from_meta( $new_size_meta );
+					wp_update_attachment_metadata( $attachment_id, $image_meta );
+				}
+			}
```

Comment:
Here you're reusing the same editor instance throughout like I mentioned before, so this is all good here. But I think for the same reason we should also do that in the block further above, where you're currently re-instantiating editors for every MIME type.

Reply:
Ah right, the way we do it here it doesn't need the initialization, we go size by size, creating each mime for each size, so the editor instance size is right. Will try removing all re-initialization and test

Reply:
Still needed:
* Verify that alternate `-scaled` and `-rotated` variants are generated from the original image
* Ensure generated filename winds up matching original, eg `filename-scaled.jpg` gets a version named `filename-scaled.webp`